[rai] Fix most tests

ao2 opened the pull request:
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Update tests to pass with recent content.

Now all tests pass, except for test_Rai_5, the one about the flv content at
http://www.rai.it/dl/RaiTV/dirette/PublishingBlock-1912dbbf-3f96-44c3-b4cf-523681fbacbc.html?channel=EuroNews

The content server now replies with an 'htm' file instead of a 'flv' one.

dstftw commented on youtube_dl/extractor/rai.py:
```diff
@@ -143,20 +142,22 @@ class RaiPlayIE(RaiBaseIE):
             'title': 'Report del 07/04/2014',
-            'alt_title': 'S2013/14 - Puntata del 07/04/2014',
-            'description': 'md5:f27c544694cacb46a078db84ec35d2d9',
+            'alt_title': 'St2013/14 - Espresso nel caff\xe8 - 07/04/2014',
```
Use plain characters not escape codes.

ao2 replied:
done

dstftw commented on youtube_dl/extractor/rai.py:
```diff
@@ -143,20 +142,22 @@ class RaiPlayIE(RaiBaseIE):
-            'season_number': 5,
+            'season_number': 6,
             'season': '2013/14',
+
```
Remove line.

ao2 replied:
done

ao2 commented:
About the escape codes, I had just pasted what the "expected" message reported when the test failed. I fixed that now, and added `# coding: utf-8` at the beginning of the file.

The extra empty line that crept in is also gone, and a `season_number` field has been updated too to make the test pass again.

Thank you for the review.

ao2 commented:
ping

iamleot commented:
@ao2 Can you please retry with current Git HEAD? After the merge of #27077 all tests should pass.

ao2 commented:
> @ao2 Can you please retry with current Git HEAD? After the merge of #27077 all tests should pass.

Almost all tests pass now, thanks.

I can only see a failure with the following one:
```
$ PYTHONPATH="$PWD/youtube-dl" python test/test_download.py TestDownload.test_Rai_4
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading webpage
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading video JSON
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading XML metadata for platform mon
ERROR: Unable to download XML: <urlopen error [SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:727)> (caused by URLError(SSLError(1, u'[SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:727)'),))
  File ".../youtube-dl/youtube-dl/youtube_dl/extractor/common.py", line 632, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File ".../youtube-dl/youtube_dl/YoutubeDL.py", line 2248, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 429, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 447, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File ".../youtube-dl/youtube_dl/utils.py", line 2736, in https_open
    req, **kwargs)
  File "/usr/lib/python2.7/urllib2.py", line 1198, in do_open
    raise URLError(err)

```

but it looks like a network error.

Feel free to close the PR if this test failure is not specific to the rai extractor.

Thank you.

remitamine commented:
```
python test_download.py TestDownload.test_Rai_4
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading webpage
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading video JSON
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading XML metadata for platform mon
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading m3u8 information
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading XML metadata for platform flash
[Rai] 3156f2f2-dc70-4953-8e2f-70d7489d4ce9: Downloading XML metadata for platform native
[info] Writing video description metadata as JSON to: test_Rai_4_3156f2f2-dc70-4953-8e2f-70d7489d4ce9.info.json
.
----------------------------------------------------------------------
Ran 1 test in 19.138s

OK
```
